src/test_helpers/namespaced.rs

```diff
@@ -116,5 +116,5 @@ pub fn initialize_namespace_tests() {
     .expect("xtables bindmount");
 
     let pid = unsafe { getpid() };
-    eprintln!("Starting test in {tmp:?}. Debug with `sudo nsenter --mount --net -t {pid}`");
+    eprintln!("Starting test in {tmp:?}. Debug with `sudo nsenter --mount --net --setuid=0 --preserve-credentials --user -t {pid}`");
```

Comment:
which wins? `--setuid=0` or `--preserve-credentials`? I thought the latter would ignore uid/gid changes?

Reply:
We want to be uid 0 in the namespace (which is **mapped** to our original ID, but is not our original id), so we override that -- preserve-credentials is for the gid really